python/res/enkf/ensemble_config.py

```diff
@@ -41,15 +43,139 @@ class EnsembleConfig(BaseCClass):
     _add_node = ResPrototype("void ensemble_config_add_node( ens_config , enkf_config_node )")
     _summary_key_matcher = ResPrototype("summary_key_matcher_ref ensemble_config_get_summary_key_matcher(ens_config)")
     _add_defined_custom_kw = ResPrototype("enkf_config_node_ref ensemble_config_add_defined_custom_kw(ens_config, char*, integer_hash)")
+    _get_trans_table = ResPrototype("void* ensemble_config_get_trans_table(ens_config)")
+    _add_summary_full = ResPrototype("void ensemble_config_init_SUMMARY_full(ens_config, char*, ecl_sum)")
 
 
+    def get_abs_path(self, file):
+        if file is not None:
+            file  = os.path.realpath(file)
+        return file
 
-    def __init__(self, config_content=None, grid=None, refcase=None):
-        c_ptr = self._alloc(config_content, grid, refcase)
+    def get_rel_path(self, file):
+        if file is not None:
+            file  = os.path.relpath(file)
+        return file
+
+    def get_filename(self, file):
+        if file is not None:
+            file  = os.path.basename(file)
+        return file
+
+    def __init__(self, config_content=None, grid=None, refcase=None, config_dict=None):
+        c_ptr = None
+        if config_dict is not None:
```

Comment:
There should be a test for using exactly one config object.

Reply:
good point, will fix this

Reply:
Actually EnsembleConfig can be created without any config object, but test for having more should be there.